### ISSUE TITLE
#bug replace datetime.date by datetime on line 801 of photosdb.py

### ISSUE THREAD

**Issue by romainjouin** [RhetTbull/osxphotos#178]
There is a bug on python 3.8 for line 801 of file photosdb.py :datetime.date should be a simple datetime
```
#self._dbphotos[uuid]["imageDate"] = datetime.date(1970, 1, 1)
self._dbphotos[uuid]["imageDate"] = datetime(1970, 1, 1)
```
See : https://stackoverflow.com/questions/62536615/mac-os-photos-programatic-access-to-the-recently-deleted-photos-folder

**Comment by RhetTbull** [comment]
@romainjouin I just pushed a fix in v0.29.29

I'll keep this issue open until I add a test -- test suite should have caught this.

In reading your stackoverflow question, I see you're trying to access photos that are in the "Recently deleted items" or trash folder.  osxphotos does not currently provide access to those because no one has ever requested it. I will open a new issue to add this feature.  I don't think it'll be terribly difficult but may take a week or two as I only work on this project as a hobby in my spare time.

As for getting access to favorites, you can do this via the PhotoInfo object:

```python
photosdb = osxphotos.PhotosDB()
photos = photosdb.photos()
favorites = [p for p in photos if p.favorite]
```

**Comment by RhetTbull** [comment]
Add test case for photo with invalid time
